experimental/proxy_attestation/src/main.rs

```diff
@@ -0,0 +1,168 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Proxy Attestation Service implementation.
+//!
+//! Proxy Attestation Service works as a Certificate Authority for TEE applications. It attests
+//! applications and creates signed certificates for TEE applications, which they use to create TLS
+//! connections with clients.
+//!
+//! The work cycle of the proxy is represented in 2 stages.
+//! First stage corresponds to the backend server attestation:
+//! - TEE application connects to the proxy
+//! - Proxy attests the application using a corresponding remote attestation protocol
+//! - Application sends the proxy a certificate signing request that contains TEE measurements
```

Comment:
I don't think the client needs to send any measurement, besides what's already in the remote attestation step. Otherwise now we have the real measurement, and also the one in this request, which seems unnecessary and confusing (and I'm sure will lead to bugs).

Reply:
+1. The measurement should be the same as in the initial attestation of the application to the proxy.